### ISSUE TITLE
Tox

### ISSUE THREAD

**Issue by ogenstad** [napalm-automation/napalm#327]
### Description of Issue/Question

There was some discussion about using Tox in the PR I created for napalm-ios. Opening an issue for that since I removed tox.ini in the PR to get it merged.

I think it's quite nice to have tox since it allows people to test the code in a very easy way, especially for libraries which are to support both 2.7 and 3.

**Comment by dbarrosop** [comment]
Can you propose a `tox.ini` file? You already had some idea : ) Would be nice a brief explanation on how to use tox with virtualenv to test all python versions.

**Comment by ogenstad** [comment]
I've added an example [here](https://github.com/napalm-automation/napalm-eos/pull/107)

The file tox.ini with comments:
```[tox]
# Python environments to run against, the ones not installed locally will be reported as failed
envlist = py27,py35

[testenv]
# Install the dependencies
deps =
    -rrequirements-dev.txt

commands=
   # Commands to run, for pylama we have to point to two directories to run against.
   # This is so that we don't run pylama against the .tox/ directory. This can also be
   # controlled in the pylama config file
   py.test --cov-report= --cov=napalm_eos test/
   pylama napalm_eos test
```

Once this file is in place someone can just `pip install tox` and run the tests from the napalm-eos root directory like below.

```
$ tox
GLOB sdist-make: /Users/patrick/src/forks/napalm-eos/setup.py
py27 create: /Users/patrick/src/forks/napalm-eos/.tox/py27
py27 installdeps: -rrequirements-dev.txt
py27 inst: /Users/patrick/src/forks/napalm-eos/.tox/dist/napalm-eos-0.5.0.zip
py27 installed: coverage==4.2,coveralls==1.1,docopt==0.6.2,future==0.16.0,Jinja2==2.8,jtextfsm==0.3.1,MarkupSafe==0.23,mccabe==0.5.2,napalm-base==0.20.4,napalm-eos==0.5.0,netaddr==0.7.18,py==1.4.31,pycodestyle==2.2.0,pydocstyle==1.1.1,pyeapi==0.7.0,pyflakes==1.3.0,pylama==7.3.1,pytest==3.0.5,pytest-cov==2.4.0,pytest-json==0.4.0,pytest-pythonpath==0.7.1,requests==2.12.3
py27 runtests: PYTHONHASHSEED='1096856675'
py27 runtests: commands[0] | py.test --cov-report= --cov=napalm_eos test/
================================================================================= test session starts =================================================================================
platform darwin -- Python 2.7.10, pytest-3.0.5, py-1.4.31, pluggy-0.4.0 -- /Users/patrick/src/forks/napalm-eos/.tox/py27/bin/python2.7
cachedir: .cache
rootdir: /Users/patrick/src/forks/napalm-eos, inifile: setup.cfg
plugins: pythonpath-0.7.1, json-0.4.0, cov-2.4.0, pylama-7.3.1
collected 36 items

test/unit/test_getters.py::TestGetter::test_get_probes_config[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_ntp_servers[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_snmp_information[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_optics[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_mac_address_table[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors_detail[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_route_to[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[issue58_neighbor_down] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_firewall_policies[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[issue68_iface_no_linkLocal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_network_instances[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_users[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[issue68_iface_no_physical_addr] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_traceroute[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue80_stacktrace_for_get_environment] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue90_veos] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_counters[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_config[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_arp_table[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_facts[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_ping[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue53] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue68_neighbor_no_local_address] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config_filtered[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_stats[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_probes_results[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_is_alive[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_peers[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_method_signatures <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED

------------------------------------------------------- generated json report: /Users/patrick/src/forks/napalm-eos/report.json --------------------------------------------------------


=============================================================================== pytest-warning summary ================================================================================
WC1 None [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.
=============================================================== 32 passed, 4 skipped, 1 pytest-warnings in 1.96 seconds ===============================================================
py27 runtests: commands[1] | pylama napalm_eos test
py35 create: /Users/patrick/src/forks/napalm-eos/.tox/py35
py35 installdeps: -rrequirements-dev.txt
py35 inst: /Users/patrick/src/forks/napalm-eos/.tox/dist/napalm-eos-0.5.0.zip
py35 installed: coverage==4.2,coveralls==1.1,docopt==0.6.2,future==0.16.0,Jinja2==2.8,jtextfsm==0.3.1,MarkupSafe==0.23,mccabe==0.5.2,napalm-base==0.20.4,napalm-eos==0.5.0,netaddr==0.7.18,py==1.4.31,pycodestyle==2.2.0,pydocstyle==1.1.1,pyeapi==0.7.0,pyflakes==1.3.0,pylama==7.3.1,pytest==3.0.5,pytest-cov==2.4.0,pytest-json==0.4.0,pytest-pythonpath==0.7.1,requests==2.12.3
py35 runtests: PYTHONHASHSEED='1096856675'
py35 runtests: commands[0] | py.test --cov-report= --cov=napalm_eos test/
================================================================================= test session starts =================================================================================
platform darwin -- Python 3.5.2, pytest-3.0.5, py-1.4.31, pluggy-0.4.0 -- /Users/patrick/src/forks/napalm-eos/.tox/py35/bin/python3.5
cachedir: .cache
rootdir: /Users/patrick/src/forks/napalm-eos, inifile: setup.cfg
plugins: pythonpath-0.7.1, json-0.4.0, cov-2.4.0, pylama-7.3.1
collected 36 items

test/unit/test_getters.py::TestGetter::test_method_signatures <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_is_alive[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_facts[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[issue68_iface_no_physical_addr] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_counters[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue80_stacktrace_for_get_environment] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue90_veos] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[issue58_neighbor_down] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors_detail[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_config[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue53] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue68_neighbor_no_local_address] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_arp_table[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_peers[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_ntp_servers[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_stats[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[issue68_iface_no_linkLocal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_mac_address_table[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_route_to[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_snmp_information[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_probes_config[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_probes_results[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_ping[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_traceroute[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_users[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_optics[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config_filtered[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_network_instances[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_firewall_policies[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED

------------------------------------------------------- generated json report: /Users/patrick/src/forks/napalm-eos/report.json --------------------------------------------------------


=============================================================================== pytest-warning summary ================================================================================
WC1 None [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.
=============================================================== 32 passed, 4 skipped, 1 pytest-warnings in 2.39 seconds ===============================================================
py35 runtests: commands[1] | pylama napalm_eos test
_______________________________________________________________________________________ summary _______________________________________________________________________________________
  py27: commands succeeded
  py35: commands succeeded
  congratulations :)

patrick at ns2803m in ~/src/forks/napalm-eos (tox) (prod)
$
```

When you run tox it creates a .tox/ directory where it installs new virtual Python environments which are used to run the tests.

**Comment by ktbyers** [comment]
Yes, this is nice...we should definitely do this.

**Comment by ktbyers** [comment]
Okay, I just tested this on napalm-ios. You accidentally dropped the [tox] header from the top of the tox.ini file.

```
[tox]
envlist = py27,py34,py35

[testenv]
deps =
    -rrequirements-dev.txt

commands=
   py.test --cov-report= --cov=napalm_ios test/
   pylama napalm_ios test
```

Also will require that anyone that runs this has the relevant Python versions on their system (I had python34, but not python35 installed).

Definitely helpful though as previously I was manually switching virtual environments which is a waste of time.

**Comment by ogenstad** [comment]
Ah, sorry. Must have been a cut & paste bug.. Of course `[tox]` should be at the top of the file.

Yes if we test against three versions of Python changes are that most people will only have one 3.x version installed. I don't think that really matters though as this would just help people with testing against 2.7 and one of the 3.x versions.

**Comment by ktbyers** [comment]
Yeah, agreed. I think it is definitely worth it.

**Comment by dbarrosop** [comment]
Looks super cool. We should have on the file whatever we want to test on travis and probably make sure travis uses it. Is it possible to also add coverage?

**Comment by dbarrosop** [comment]
Something I don't see there is `pylama`. Can it be added? @mierdin added this `tox.ini` and some other tooling long ago. We didn't merge it but I think we should review it for ideas:

https://github.com/napalm-automation/napalm-junos/pull/37/files#diff-b91f3d5bd63fcd17221b267e851608e8

**Comment by ogenstad** [comment]
Yeah Travis can call tox too. 

Regarding coverage that can be added too. I change the line in setup.cfg `addopts = --cov=./ -vs` to `addopts = -vs` (so that the virtualenv doesn't get included). Than changed the command in tox to: `py.test --cov=napalm_eos --cov-report term-missing``

For pylama it was just silent since it didn't have anything to complain about. I've added a few extra lines to a file to make it complain below.

```
$ tox
GLOB sdist-make: /Users/patrick/src/forks/napalm-eos/setup.py
py27 inst-nodeps: /Users/patrick/src/forks/napalm-eos/.tox/dist/napalm-eos-0.5.0.zip
py27 installed: coverage==4.2,coveralls==1.1,docopt==0.6.2,future==0.16.0,Jinja2==2.8,jtextfsm==0.3.1,MarkupSafe==0.23,mccabe==0.5.2,napalm-base==0.20.4,napalm-eos==0.5.0,netaddr==0.7.18,py==1.4.31,pycodestyle==2.2.0,pydocstyle==1.1.1,pyeapi==0.7.0,pyflakes==1.3.0,pylama==7.3.1,pytest==3.0.5,pytest-cov==2.4.0,pytest-json==0.4.0,pytest-pythonpath==0.7.1,requests==2.12.3
py27 runtests: PYTHONHASHSEED='243346193'
py27 runtests: commands[0] | py.test --cov=napalm_eos --cov-report term-missing
================================================================================= test session starts =================================================================================
platform darwin -- Python 2.7.10, pytest-3.0.5, py-1.4.31, pluggy-0.4.0 -- /Users/patrick/src/forks/napalm-eos/.tox/py27/bin/python2.7
cachedir: .cache
rootdir: /Users/patrick/src/forks/napalm-eos, inifile: setup.cfg
plugins: pythonpath-0.7.1, json-0.4.0, cov-2.4.0, pylama-7.3.1
collected 36 items

test/unit/test_getters.py::TestGetter::test_get_interfaces_counters[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_firewall_policies[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_config_filtered[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_is_alive[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_probes_results[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_ping[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_mac_address_table[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_peers[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_network_instances[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue80_stacktrace_for_get_environment] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue90_veos] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_probes_config[no_test_case_found] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_facts[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_snmp_information[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_route_to[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[issue58_neighbor_down] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_servers[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue53] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue68_neighbor_no_local_address] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_optics[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_config[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_stats[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors_detail[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_traceroute[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_arp_table[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_users[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[issue68_iface_no_physical_addr] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[issue68_iface_no_linkLocal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[normal] <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_method_signatures <- .tox/py27/lib/python2.7/site-packages/napalm_base/test/getters.py PASSED

------------------------------------------------------- generated json report: /Users/patrick/src/forks/napalm-eos/report.json --------------------------------------------------------

---------- coverage: platform darwin, python 2.7.10-final-0 ----------
Name                           Stmts   Miss  Cover   Missing
------------------------------------------------------------
napalm_eos/__init__.py             8      2    75%   21-22
napalm_eos/eos.py                762    154    80%   76, 82-83, 99-102, 105, 110-114, 126-162, 166, 170, 174-182, 186-193, 198-202, 206-209, 254-258, 321-322, 362, 393-394, 504, 548-571, 638, 661, 668-669, 676, 696-697, 702, 720, 724, 760, 775-777, 781, 795-796, 849, 865-866, 877-881, 912, 921, 924, 926, 965-966, 995-997, 1006, 1071-1079, 1094, 1106, 1121, 1181, 1187, 1199-1200, 1208, 1337-1349, 1362-1367, 1462, 1479-1481, 1494, 1546-1548, 1577, 1580, 1596
napalm_eos/utils/__init__.py       0      0   100%
------------------------------------------------------------
TOTAL                            770    156    80%

=============================================================================== pytest-warning summary ================================================================================
WC1 None [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.
=============================================================== 32 passed, 4 skipped, 1 pytest-warnings in 0.32 seconds ===============================================================
py27 runtests: commands[1] | pylama napalm_eos test
napalm_eos/eos.py:55:1: E303 too many blank lines (5) [pep8]
ERROR: InvocationError: '/Users/patrick/src/forks/napalm-eos/.tox/py27/bin/pylama napalm_eos test'
py35 inst-nodeps: /Users/patrick/src/forks/napalm-eos/.tox/dist/napalm-eos-0.5.0.zip
py35 installed: coverage==4.2,coveralls==1.1,docopt==0.6.2,future==0.16.0,Jinja2==2.8,jtextfsm==0.3.1,MarkupSafe==0.23,mccabe==0.5.2,napalm-base==0.20.4,napalm-eos==0.5.0,netaddr==0.7.18,py==1.4.31,pycodestyle==2.2.0,pydocstyle==1.1.1,pyeapi==0.7.0,pyflakes==1.3.0,pylama==7.3.1,pytest==3.0.5,pytest-cov==2.4.0,pytest-json==0.4.0,pytest-pythonpath==0.7.1,requests==2.12.3
py35 runtests: PYTHONHASHSEED='243346193'
py35 runtests: commands[0] | py.test --cov=napalm_eos --cov-report term-missing
================================================================================= test session starts =================================================================================
platform darwin -- Python 3.5.2, pytest-3.0.5, py-1.4.31, pluggy-0.4.0 -- /Users/patrick/src/forks/napalm-eos/.tox/py35/bin/python3.5
cachedir: .cache
rootdir: /Users/patrick/src/forks/napalm-eos, inifile: setup.cfg
plugins: pythonpath-0.7.1, json-0.4.0, cov-2.4.0, pylama-7.3.1
collected 36 items

test/unit/test_getters.py::TestGetter::test_method_signatures <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_is_alive[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_facts[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[issue68_iface_no_physical_addr] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_counters[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue80_stacktrace_for_get_environment] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[issue90_veos] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_environment[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[issue58_neighbor_down] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_lldp_neighbors_detail[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_config[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue53] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[issue68_neighbor_no_local_address] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_bgp_neighbors_detail[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_arp_table[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_peers[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_ntp_servers[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_ntp_stats[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[issue68_iface_no_linkLocal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_interfaces_ip[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_mac_address_table[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_route_to[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_snmp_information[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_probes_config[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_get_probes_results[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED
test/unit/test_getters.py::TestGetter::test_ping[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_traceroute[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_users[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_optics[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_config_filtered[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_network_instances[normal] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py PASSED
test/unit/test_getters.py::TestGetter::test_get_firewall_policies[no_test_case_found] <- .tox/py35/lib/python3.5/site-packages/napalm_base/test/getters.py SKIPPED

------------------------------------------------------- generated json report: /Users/patrick/src/forks/napalm-eos/report.json --------------------------------------------------------

---------- coverage: platform darwin, python 3.5.2-final-0 -----------
Name                           Stmts   Miss  Cover   Missing
------------------------------------------------------------
napalm_eos/__init__.py             8      2    75%   21-22
napalm_eos/eos.py                762    154    80%   76, 82-83, 99-102, 105, 110-114, 126-162, 166, 170, 174-182, 186-193, 198-202, 206-209, 254-258, 321-322, 362, 393-394, 504, 548-571, 638, 661, 668-669, 676, 696-697, 702, 720, 724, 760, 775-777, 781, 795-796, 849, 865-866, 877-881, 912, 921, 924, 926, 965-966, 995-997, 1006, 1071-1079, 1094, 1106, 1121, 1181, 1187, 1199-1200, 1208, 1337-1349, 1362-1367, 1462, 1479-1481, 1494, 1546-1548, 1577, 1580, 1596
napalm_eos/utils/__init__.py       0      0   100%
------------------------------------------------------------
TOTAL                            770    156    80%

=============================================================================== pytest-warning summary ================================================================================
WC1 None [pytest] section in setup.cfg files is deprecated, use [tool:pytest] instead.
=============================================================== 32 passed, 4 skipped, 1 pytest-warnings in 0.42 seconds ===============================================================
py35 runtests: commands[1] | pylama napalm_eos test
napalm_eos/eos.py:55:1: E303 too many blank lines (5) [pep8]
ERROR: InvocationError: '/Users/patrick/src/forks/napalm-eos/.tox/py35/bin/pylama napalm_eos test'
_______________________________________________________________________________________ summary _______________________________________________________________________________________
ERROR:   py27: commands failed
ERROR:   py35: commands failed

patrick at ns2803m in ~/src/forks/napalm-eos (tox●) (prod)
$
```

**Comment by dbarrosop** [comment]
Regarding pylama, I deleted my own comment because I realized it was there and I had just missed it : (

**Comment by Mierdin** [comment]
FWIW the benefit of using something like Tox is to simplify the travis
configuration enough that it cuts down on what's needed for the user to run
the same tasks that travis would, on their laptop. That's the goal anyways

*Matt Oswalt*
Network Software Engineer
(c) 513.202.3138
(e) matt@keepingitclassless.net
(t) @Mierdin <http://twitter.com/mierdin>
(w) http://keepingitclassless.net/

**Comment by dbarrosop** [comment]
I commented on the PR (https://github.com/napalm-automation/napalm-eos/pull/107). I personally think we should move forward with this although I'd like to make sure it integrates nicely with travis.

**Comment by dbarrosop** [comment]
@napalm-automation/council would you mind taking a look to this? https://github.com/napalm-automation/napalm-eos/pull/107

**Comment by mirceaulinic** [comment]
Closing this, as I think this will be solved anyway during the reunification.
